### ISSUE TITLE
[Slack] OAuth v2 with new bot scopes is not working after redirect

### ISSUE THREAD

**Issue by jmnsf** [microsoft/BotFramework-Services#179]
Slack recently added more granular scopes and deprecated the catch-all "bot" scope that BotFramework uses by default. It also added a new oauth endpoint to use these new scopes.

I can generate a new add-to-slack link and complete the oauth flow, for example:

```
https://slack.com/oauth/v2/authorize?scope=app_mentions:read,chat:write,emoji:read,im:history,im:read,im:write,links:read,links:write,mpim:history,mpim:read,reactions:read,reactions:write,team:read,users.profile:read,users:write&user_scope=team:read&client_id=888888888888.9999999999&redirect_uri=https%3a%2f%2fslack.botframework.com%2fHome%2fauth&state=xx-uu-yy-zz
```

But on redirect the user sees:
![image](https://user-images.githubusercontent.com/6831662/73841928-8881ae00-4813-11ea-86da-c2837f3ecef1.png)
(This is at `https://slack.botframework.com/Home/auth?code=123123.123123213.123123.123123&state=xx-uu-yy-zz`)

And BF does not send an `installationUpdate` to the bot.

**Comment by jwiley84** [comment]
Let me look into this, and if need be, pass this up to someone to get fixed.

**Comment by jwiley84** [comment]
I'm still working on clarifying the instructions to add a bot to slack properly, but I am unable to repro your issue. I need a few more details from you:

My URL is as follows:

https://slack.com/oauth/authorize?scope=incoming-webhook&client_id=MYCLINTID"><img alt=""Add to Slack"" height="40" width="139" src="https://platform.slack-edge.com/img/add_to_slack.png" srcset="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x


What happens if you remove all those scopes from you 'Add' button?

What scopes  have you added to your app on the Slack API page, under 'Oauth & Permissions'?

**Comment by jmnsf** [comment]
Hi @jwiley84, thanks for looking into this. Note that the new scopes are now bound to a new OAuth URL, `https://slack.com/oauth/v2/authorize`. This means also that BotFramework needs to call the `oauth.v2.access` method for exchanging the OAuth code for an Auth Token (I believe this is the part that fails since you're likely to be calling `oauth.access` instead).

I've worded the issue as a bug, but it's more likely to be a feature request for support of the new Slack OAuth flows.

**Comment by arturl** [comment]
Hi @jmnsf, we understand the issue but don't have a date on when we do this work. We're putting it on the backlog. Thanks for reporting it.

**Comment by sremiger1** [comment]
> Hi @jmnsf, we understand the issue but don't have a date on when we do this work. We're putting it on the backlog. Thanks for reporting it.

Can you summarize a work around that can be used?  We are in the same position.

**Comment by jmnsf** [comment]
@sremiger1 I'm afraid I bit the bullet and just integrated with Slack directly. With this being a blocker on getting an app published on the Slack App Directory and not having access to the Blocks API I just went ahead and made the switch.

I'm not using any of the botframework SDKs (just the REST API) so that made it a lot easier I think.

**Comment by arturl** [comment]
We have a fix, will be deployed soon. ETA end of May.

**Comment by arturl** [comment]
The fix has been deployed. We will update the documentation soon.

**Comment by arturl** [comment]
Closed prematurely. The fix is still in development and will be deployed by 5/11.
